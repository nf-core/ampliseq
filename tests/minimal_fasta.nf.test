nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "nf-core/ampliseq"

    test("Minimal Fasta Input") {

        options "-profile test_fasta,docker"

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                  path("$outputDir/barrnap/rrna.bac.gff"),
                                  path("$outputDir/barrnap/rrna.euk.gff"),
                                  path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/dada2/ref_taxonomy.txt")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax.tsv").exists() },  
                { assert new File("$outputDir/dada2/ASV_tax_species.tsv").exists() },
                { assert snapshot(path("$outputDir/input/ASV_seqs.fasta")).match("input") }           
            )
        }

    }

}
