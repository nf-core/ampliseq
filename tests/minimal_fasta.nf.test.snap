{
    "input": {
        "content": [
            "ASV_seqs.fasta:md5,39b0d4ad96b37fa68d4ee5c21b667fed"
        ],
        "timestamp": "2023-04-15T21:02:11+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,fa3e1f3ed59ef3771ec4744ae3762e8a"
        ],
        "timestamp": "2023-04-15T21:02:11+0000"
    },
    "dada2": {
        "content": [
            "ref_taxonomy.txt:md5,815c4fce9f3d1de019fb995a43fb66ed"
        ],
        "timestamp": "2023-04-15T21:02:11+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6dae470aace9293d5eb8c318584852dd",
            "rrna.bac.gff:md5,439a9084f089120f700f938dfb58fa41",
            "rrna.euk.gff:md5,c9bc1d9d8fb77dc19c95dee2d53840eb",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-04-15T21:02:11+0000"
    }
}